File: src/main/java/br/com/blz/testjava/controller/dto/InventoryDTO.java
package br.com.blz.testjava.controller.dto;

import java.util.*;

public class InventoryDTO {
    private final List<WarehouseDTO> warehouses = new ArrayList<>();

    public void add(final WarehouseDTO item) {
        warehouses.add(item);
    }

    public void addAll(final Collection<WarehouseDTO> items) {
        warehouses.addAll(items);
    }

    public List<WarehouseDTO> getWarehouses() {
        return Collections.unmodifiableList(warehouses);
    }

    public Integer getQuantity() {
        return warehouses.stream()
            .reduce(
                0,
                (partialQuantity, warehouse) -> partialQuantity + warehouse.getQuantity(),
                Integer::sum);
    }
}

File: src/main/kotlin/br/com/blz/testjava/controller/dto/InventoryDTO.kt
package br.com.blz.testjava.controller.dto

import br.com.blz.testjava.domain.entities.Inventory

class InventoryDTO() {
  var warehouses: List<WarehouseDTO> = listOf()
  var quantity: Int = 0

  constructor(inventory: Inventory): this() {
    quantity = inventory.getQuantity()
    warehouses = inventory.warehouses.map { warehouse -> WarehouseDTO(warehouse) }
  }

}
